kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ingress/IngressDefinition.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator.ingress;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Stateless Ingress model that:
+ * <ol>
+ *     <li>describes the requirements of the ingress (e.g. how many exclusive ports it requires)</li>
+ *     <li>can instantiate IngressInstances, allowing multiple virtual clusters to use the same IngressDefinition</li>
+ * </ol>
+ */
+public interface IngressDefinition {
+
+    /**
+     * The raw resource that was translated into this definition
+     * @return resource
+     */
+    @NonNull
+    KafkaProxyIngress resource();
+
+    /**
+     * Create an instance of the Ingress with identityful ports allocated to it. Identityful meaning that
```

Comment:
I was trying to find a better term than `exclusive` port like we use in the proxy because inclusive/exclusive is confusing language when we're describing a range.

Reply:
"Identifying Port" might make for less wincing.